platform/linux-generic/pool/subsystem.c

```diff
@@ -15,3 +19,219 @@ ODP_SUBSYSTEM_CONSTRUCTOR(pool)
 	odp_subsystem_constructor(pool);
 }
 
+int odp_pool_init_global(void)
+{
+	pool_module_t *mod;
+
+	mod = odp_subsystem_active_module(pool, mod);
+	if (mod == NULL) {
+		ODP_ERR("No active module in pool subsystem\n");
+		return -1;
+	}
+	if (mod->base.init_global == NULL) {
+		ODP_ERR("No defined init_global function "
+			"in module %s of pool subsystem\n", mod->base.name);
+		return -1;
+	}
```

Comment:
It makes sense to do these checks at init time when dynamic loading is involved, but once a module is loaded and bound these just add overhead, which should be unnecessary. The `odp_pool_xxx()` APIs are not executed with great frequency so this overhead may not matter here, but for things like pktio operations such extra overhead would be very undesirable.